org.eclipse.scanning.api/src/org/eclipse/scanning/api/device/IAttributableDevice.java

```diff
@@ -24,18 +25,26 @@
 public interface IAttributableDevice {
 
 	/**
-	 * Gets the an attribute on the device
+	 * Gets the attribute on the device with the given name.
+	 * @param attributeName name of attribute
+	 * @return the attribute with the given name
+	 * @throws ScanningException if the attribute cannot be retrieved for any reason
 	 */
-	public Object getAttribute(String attribute) throws ScanningException;
+	public <T> IDeviceAttribute<T> getAttribute(String attributeName) throws ScanningException;
 	
 	/**
-	 * Gets a list of all attributes on the device
+	 * Gets a list of all attributes on the device.
+	 * @return all attributes
+	 * @throws ScanningException if the attributes cannot be retrieved for any reason 
 	 */
-	public <A> List<A> getAllAttributes() throws ScanningException;
+	public List<IDeviceAttribute<?>> getAllAttributes() throws ScanningException;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1452)